transaction/context.go

```diff
@@ -0,0 +1,45 @@
+package transaction
+
+import (
+	"context"
+	"fmt"
+)
+
+type contextKey int
+
+const (
+	trackedKey contextKey = iota
+	transactionIDKey
```

Comment:
To prevent collisions every item tracked in a context should be managed by its own package. The key type is always private and having it in its own package makes it unique by design. This is the concept encouraged by the golang folks. See https://blog.golang.org/context and https://golang.org/pkg/context/#Context. Mixing the context management with other implementations is wrong IMO. What would make sense to me would be to have dedicated packages like these:

- `transaction/trackedctx` with methods like `FromContext` and `WithContext`. 
- `transaction/idctx` with methods like `FromContext` and `WithContext`.